prow/git/v2/client_factory.go

```diff
@@ -339,21 +346,69 @@ func (c *clientFactory) ClientForWithRepoOpts(org, repo string, repoOpts RepoOpt
 		// something unexpected happened
 		return nil, err
 	} else {
-		// we have cloned the repo previously, but will refresh it
-		if err := cacheClientCacher.RemoteUpdate(); err != nil {
-			return nil, err
+		// We have cloned the repo previously, but will refresh it. By default
+		// we refresh all refs with a call to `git remote update`.
+		//
+		// This is the default behavior if FetchCommits is empty or nil (i.e.,
+		// when we don't define a targeted list of commits to fetch directly).
+		if len(repoOpts.FetchCommits) == 0 {
+			if err := cacheClientCacher.RemoteUpdate(); err != nil {
+				return nil, err
+			}
 		}
```

Comment:
We still ignore `NoFetchTags` in this case. We should at least document this in the GoDoc for the field. https://github.com/kubernetes/test-infra/pull/30453#discussion_r1302255762

Reply:
The flag is deprecated in https://github.com/kubernetes/test-infra/pull/30428 (https://github.com/kubernetes/test-infra/pull/30428/commits/5c0e7eca5c89f04b9dfdf5b158202bedb77bd0b2).